### ISSUE TITLE
Booster vaccination should be shown as such, not 2/2 for Janssen.

### ISSUE THREAD

**Issue by SoftwareApe** [corona-warn-app/cwa-app-android#4473]
## Avoid duplicates
* [ x] This enhancement request has not already been raised before
* [ x] Enhancement request is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in [CWA-Wishlist](https://github.com/corona-warn-app/cwa-wishlist)
* [ ] If you are proposing a new feature, please do so in [CWA-Wishlist](https://github.com/corona-warn-app/cwa-wishlist)

## Current Implementation
With Janssen, Booster is shown as 2/2. In some states (e.g. BW) 2G+ means no need to have a test certificate.

With Janssen the Booster gets easily confused with the final normal vaccination. This makes it necessary to explain everywhere that the first vaccination was Janssen, thus showing 1/1, and therefore 2/2 is the booster.

## Suggested Enhancement
Since we don't know how many boosters there are going to be it doesn't make sense to show x/y.

Instead it should be 1st booster, 2nd Booster, ...

## Expected Benefits
Less hassle explaining things. Less confusion. Less interaction with people.

**Comment by Ein-Tim** [comment]
The EU dcc system requires this x/y style of showing. If we change this, we break compatibility with other EU countries. 

Still, I understand your situation pretty well and think this should be brought to the attention of the stakeholders (RKI/BMG) so that they can discuss what the best solution is.

**Comment by SoftwareApe** [comment]
Could you then say 2/2 (booster)? Or something to make it clear that this is the booster?

**Comment by SoftwareApe** [comment]
That wouldn't remove the required style of x/y

**Comment by Ein-Tim** [comment]
@SoftwareApe 

Not sure but AFAIK it's not possible to add other text to this field without breaking compatibility with other EU countries.

However, I'm no expert in regard to DCCs and I'm sure @vaubaehn could comment better on this (-:

**Comment by SoftwareApe** [comment]
Thank you, Tim!

**Comment by vaubaehn** [comment]
@SoftwareApe 
Yes, @Ein-Tim is correct: the fields for the dose number and the total number of doses in a series are defined to be integer (positive numbers) solely:
https://github.com/ehn-dcc-development/ehn-dcc-schema/blob/a603410d760fefc9073931c8c807759d9714c136/DCC.combined-schema.json#L88-L93

**Comment by dsarkar** [comment]
@SoftwareApe See also https://www.coronawarn.app/de/faq/#vac_booster_jj

**Comment by SoftwareApe** [comment]
@dsarkar that's not really related to this discussion. I just want the booster vaccination to be marked as such. I don't care about the 14 day windoow which is off course incorrect, but could be recognized by the app from the available information.

It's just bad that those vaccinated with J&J are forgotten alll the time even though it's not even what I would call an edge case.

You could e.g. show 2/2 (booster) and comply with the guidelines from RKI and make it clear that this was the booster shot.

**Comment by Ein-Tim** [comment]
> It's just bad that those vaccinated with J&J are forgotten all the time even though it's not even what I would call an edge case.

It's also everybody who received a booster after their vaccination after they recovered.

> You could e.g. show 2/2 (booster) and comply with the guidelines from RKI and make it clear that this was the booster shot.

As @vaubaehn explained, this is not possible. 

Within the EU DCC schema, I guess, the only thing we could do is add the information to the app. 

But when should it be shown? If a J&J certificate and a 2/2 certificate is added to the app? 

Okay, this is possible. But if somebody recovered, was vaccinated one time (1/1) and then another time (2/2), we would have to check for one recovery, one 1/1 and one 2/2 certificate in the app. 
If one of the certificates is missing (which is most likely because recovery certificate get invalid 6 months after the recovery), there is no way for the app to tell that this is a booster vaccination on a vaccination after recovery.

We could only show a general note If we detect"2/2" certificate is added on top of a "1/1" certificate. 



But one good thing: In all these edge cases, users have a valid certificate, so nobody will have to stay outdoors (-:

**Comment by SoftwareApe** [comment]
>We could only show a general note if we detect a "2/2" certificate is added on top of a "1/1" certificate.

Yes, I think this should do it. :)

**Comment by Cybso** [comment]
> > We could only show a general note if we detect a "2/2" certificate is added on top of a "1/1" certificate.
> 
> Yes, I think this should do it. :)

This would show a hint in CWA, but other apps like CovPassCheck will still fail to verify the certificate, because they do not now about the first one.

I think the only workaround that would not require a change in the European standard would be to issue booster vaccines as 3/3 in principle, regardless of the number of vaccinations in the first series. But that would probably require an official directive from the Ministry of Health so that issuers would not be liable to prosecution for falsification.

**Comment by vaubaehn** [comment]
@Cybso 
> > > We could only show a general note if we detect a "2/2" certificate is added on top of a "1/1" certificate.
> > 
> > 
> > Yes, I think this should do it. :)
> 
> This would show a hint in CWA, but other apps like CovPassCheck will still fail to verify the certificate, because they do not now about the first one.

Seems true, this is why I proposed https://github.com/Digitaler-Impfnachweis/covpass-android/issues/107

> I think the only workaround that would not require a change in the European standard would be to issue booster vaccines as 3/3 in principle, regardless of the number of vaccinations in the first series. But that would probably require an official directive from the Ministry of Health so that issuers would not be liable to prosecution for falsification.

In my understanding, issuing booster vaccines as 3/3 would actually require a change in the European standard on how to count vaccination series... In all the EU that would need to be common agreement, not only in Germany.

**Comment by SoftwareApe** [comment]
>This would show a hint in CWA, but other apps like CovPassCheck will still fail to verify the certificate, because they do not now about the first one.

I mean the two weeks time slot is a separate issue. I would like the user to instantly see that it's a booster shot. That would only need some extra text to clarify.

**Comment by SoftwareApe** [comment]
BTW at least CovPass is working on the 14 day issue it seems.

>Da es sich bei diesem Umstand offensichtlich um einen technischen Fehler handelt, muss in absehbarer Zeit eine Lösung nachgereicht werden. Eine Servicemitarbeiterin der CovPass-App bestätigte uns am Telefon, dass die Entwickler bereits an einer Lösung arbeiten, sodass die App zwischen Zweitimpfung und Auffrischungsimpfung unterscheiden kann.

https://www.stuttgarter-zeitung.de/inhalt.booster-nach-johnson-ungueltig-mhsd.9c9a27cc-2866-4d1f-b93d-f5a622ab47b0.html

**Comment by Ein-Tim** [comment]
@SoftwareApe 

Not too sure how they will fix this. Maybe you could open an issue in the CovPass repo and ask?

**Comment by bigboipete** [comment]
Couldn't this be solved by sticking to the x/y scheme regarding the required data set, but additionally add a "booster" label (or whatever suits there technically) on app UI level. Given that it could be reliably evaluated within the context of the underlying data, whether it's a booster or not.

Nobody said the booster info has to be added to the fixed and defined data set. Or did I miss something?

**Comment by SoftwareApe** [comment]
@bigboipete exactly

**Comment by chrisv2** [comment]
FWIW, this issue seems to gain more public attention recently. What's interesting is that the info seems to be there, it's just not shown properly:

> Wenn der Apotheker das Impfzertifikat ausstellt, klickt er in diesen Fällen in einem Pull-Down-Menü „2/2 Booster“ an. Im Ergebnis jedoch verstehen die Apps seine 2/2-Eingabe nicht als Boosterung, sondern als normale zweite Impfung. 

https://bnn.de/karlsruhe/karlsruhe-stadt/fehler-apps-digitales-impfzertifikat-booster-johnson-impfungen-genesene-corona-warnapp-covass-apotheken

> Das Sozialministerium in Stuttgart hat am Donnerstag auf die berichteten Schilderungen von technischen Problemen von Geboosterten nach der Genesung oder einer Erstimpfung mit Johnson & Johnson reagiert: Das Problem sei bekannt, das Ministerium erwarte vom Bund, die technischen Voraussetzungen für eine Kontrolle der Impfzertifikate zu schaffen.

https://bnn.de/nachrichten/politik/corona-impfung-fehlermeldungen-booster-zertifikat-johnson-johnson-genesene-2g-plus-kontrollen-sozialministerium-bund

**Comment by dsarkar** [comment]
Dear @chrisv2 @SoftwareApe @bigboipete, dear community,

we are aware of this issue. We will update the FAQ explaining the reasons. In short, the way the JSON scheme was definded on EU level, does not allow to map correctly these above mentioned cases. One could imagine to show it correctly in the CWA, since the app can see all certificates in the "wallet". However, since only one certificate is verified with a CheckApp, this would not be an entirely satisfying solution. 

A proper solution would be to adapt the definition of the JSON on EU level, or find a national solution.

**Comment by bigboipete** [comment]
To sum this up for my understanding: A "Booster" label to display within the app wouldn't be a problem, but there wouldn't be any benefit in case of an "external" status check, because a correct booster identification from "outside" is not possible yet due to the given data set, right?

**Comment by jfhaecker** [comment]
Looking at the json schema v1.3.0 this looks interesting. Just scanned my own Moderna-Booster-certficate after Johnson vaccination with vacdec: dn=2 and sd=2. Seems pretty hard for an app to see that this should be a booster. 
Semantics of dn/sd seems weird. Maybe we should set sd=3, it's a booster, baby.

**Comment by Panda-vid** [comment]
Sorry that I am late to the party but I think the issue with checking on top of previous certificates is that the validation via QR would still not work since the QR code of neither the original dcc and the booster dcc contain information regarding the other vaccination as mentioned before.

> Looking at the json schema v1.3.0 this looks interesting. Just scanned my own Moderna-Booster-certficate after Johnson vaccination with vacdec: dn=2 and sd=2. Seems pretty hard for an app to see that this should be a booster. Semantics of dn/sd seems weird. Maybe we should set sd=3, it's a booster, baby.

This might be the only solution even though it is kind of a hack. 

I would change the data of the certificate such that it contains the date at which the certificate is valid. 
I feel like the app should not be able to decide when a certificate is valid anyways. It should only be able to verify a dcc is genuine and then just read the information in the it. Obviously, I understand this is not possible anymore or at least it would be hard to implement.

Another help would be if the issuer would use data from the last certificate by requireing the doctor etc. to scan the last dcc. At this stage the issuer could automatically mark the new cert as 3/3 (or the correct combination).

**Comment by Ein-Tim** [comment]
FYI: https://ec.europa.eu/commission/presscorner/detail/en/ip_21_6837 says:

"Boosters will be recorded as follows:

3/3 for a booster dose following a primary 2-dose vaccination series.
2/1 for a booster dose following a single-dose vaccination or a one dose of a 2-dose vaccine administered to a recovered person."

**Comment by jfhaecker** [comment]
thx, thats fun as some federal states(BY) seem to announce that a dose after a single-dose vaccination (Janssen in very most cases) is not a "booster" but an optimization of the basic immunization. This seems to be based on the current STIKO recommendation. 
If that's true the setting 2/2 for a dose after single-dose vaccination seems to be correct. After 3 month those people can get a real booster and are back in the game with 3/3.
In BW they announced the rule that you are 2G+ without a daily test if you have a booster (3/3) or your last dose was within the last 3 month.
Sadly, with this mixture of EU, member state and federal state rules there seems no easy way to have an algorithm which correctly detects the "boostered"-state in all cases.

**Comment by ElpadoCan** [comment]
The Bavaria Ministry for Health actually announced that JJ + mRNA doesn't count as 2G+, but the law didn't change. The law still states that with JJ one is fully vaccinated and the second dose is the an additional dose to the fully vaccinated status. Am I missing something here? Are they announcing new rules before even having the regulation written down?

**Comment by DerVogel2020** [comment]
@ElpadoCan I read the same yesterday in an article, we below. If you read the Stiko recommendation, the mRNA vaccination after a JJ is not called "Auffrischungsimpfung"  so you could interpret it that way.

https://www1.wdr.de/nachrichten/themen/coronavirus/booster-dritte-impfung-johnson-100.html
https://www.fitbook.de/news/wann-gelten-johnson-johnson-geimpfte-als-geboostert


Interesting will be, what numbers will be in the DCC...

**Comment by ElpadoCan** [comment]
@DerVogel2020 yes, I know the STIKO recommendation, but it's just a recommendation as long as the law isn't updated. The law that you can find [here](https://www.gesetze-bayern.de/Content/Document/BayIfSMV_15-4) still says:

_"geimpfte Personen im Sinne des § 2 Nr. 2 SchAusnahmV die zusätzlich eine weitere Impfstoffdosis als Auffrischungsimpfung erhalten haben nach Ablauf von 14 Tagen nach dieser Impfung, soweit nicht bundesrechtlich anderes geregelt ist."_

and if you go to check the definition of fully vaccinated as stated in § 2 Nr. 2 SchAusnahmV ([here](https://www.gesetze-im-internet.de/schausnahmv/__2.html)) you can read:

_"entweder aus einer vom Paul-Ehrlich-Institut im Internet unter der Adresse www.pei.de/impfstoffe/covid-19 veröffentlichten Anzahl von Impfstoffdosen, die für eine vollständige Schutzimpfung erforderlich ist, besteht und seit der letzten erforderlichen Einzelimpfung mindestens 14 Tage vergangen sind oder"_

and of course, when you go to check which vaccine and how many doses count as fully vaccinated according to the Paul-Ehrlich-Institut (link [here](https://www.pei.de/DE/newsroom/dossier/coronavirus/coronavirus-inhalt.html?nn=169730&cms_pos=3)) you find the **JJ vaccine with 1 dose**.

So my question is, what am I missing? Every media and even the FAQ on the bavarian government website clearly states that JJ+1 dose doesn't count but the law is not updated yet? How is that possible?

If this was true, they need to find a way to get people with JJ to catch up with the others. Because by the time I am eligible for my third dose after my last booster, they will change the rules and the "boostered" definition will state that you need a 4th dose and so forth. This way JJ people will never catch up, which is total nonsense.

**Comment by haskamp-1** [comment]
Liebe Entwickler,

Ab morgen gilt in Hamburg 2G+. Heute habe ich eine Kellnerin gefragt, ob sie meinen Impfstatus (J&J mit BioNTech ein paar Monate später) anerkennt, weil dies ja als geboostert gilt:
https://www.ndr.de/nachrichten/hamburg/coronavirus/Corona-Diese-Regeln-gelten-momentan-in-Hamburg,coronavirus2248.html

Sie verneinte das, weil dort nicht 3/3 steht.

Ich zweifle ja gar nichts an, aber hier zu lesen, dass es nicht irgendwelchen EU Regeln entspricht ist lächerlich. Hier geht es um ganz praktische Probleme im Leben und man sollte sich mal als ITler überlegen, warum er überhaupt den Auftrag zur Programmierung von CovPass bekommen hat - garantiert war es nicht der Auftrag den Geimpften und Genesenen das Leben schwer zu machen. Entweder in CovPass steht bei J&J geboostert dahinter oder es muss dort 3/3 stehen. Alles andere verstehen die Leute da draußen nicht, die haben nicht alle einen Doktortitel! Ich empfinde die anlehnenden Antworten als Frechheit.

Viele Grüße 
Heinrich Haskamp

**Comment by DerVogel2020** [comment]
Die anzeigten Zahlen stammen aus dem Zertifikat, das die Apotheke (oder der Arzt) erstellt hat.
Da können die Entwickler nichts dran ändern.

**Comment by haskamp-1** [comment]
Die App kann nicht erkennen, dass die erste Impfung J&J und die zweite ein RNA Impfstoff ist, um dann stumpft „geboostert“ hinzuschreiben, damit ein Türsteher es auf Anhieb erkennt?
Wenn das wirklich so ist, dann haben wir wirklich Fachkräftemangel… das wäre peinlich!

Viele Grüße 
Heinrich Haskamp

**Comment by ElpadoCan** [comment]
@haskamp-1 Sie hätten darum bitten sollen, mit einem Manager zu sprechen. Dann hättest du ihnen den Link zum Gesetz und den Link des Paul-Einrich-Instituts zeigen sollen, das JJ als vollständig geimpft betrachtet. Wenn man Ihnen dann immer noch nicht zuhört, hätten Sie damit drohen sollen, die Polizei zu rufen. Ich habe das gestern in einem Schwimmbad gemacht und es hat funktioniert. Sie gaben schließlich auf und erlaubten mir den Eintritt. 

Um ehrlich zu sein, ist das wirklich lächerlich, und ich bin sicher, dass ein Anwalt oder Journalist viel Spaß dabei haben wird, darüber zu berichten.

Aber im Allgemeinen stimme ich zu. So wie es aussieht, ist das Zertifikat mit 2/2 in der App völlig nutzlos. Ich bringe eine Kopie des Gesetzes mit und alle meine Zeugnisse, was im Grunde ein totales Versagen des App-Projekts ist (obwohl es nicht die Schuld der Entwickler ist).

**Comment by dsarkar** [comment]
Hi @haskamp-1,

Die Entwickler setzen um, was Ihnen in Auftrag geben wird. Das Problem liegt nicht an der App, sondern an der Definition der Nummerierung. Siehe https://www.coronawarn.app/de/faq/#vac_booster_jj


Viele Grüsse, DS

---
Corona-Warn-App Open Source Team
